### PR TITLE
Do not update `estimate_price` of a variant that is not available for the shop on OrderCycle update

### LINKED ISSUE #9292: Changes to incoming/outgoing products lists triggering 'failed to update order cycle' error
## Description

Banc Organics are seeing an error when they try to make changes to their incoming and outgoing product lists on order cycles with schedules attached. The error 'Failed to update order cycles' is triggered _but_ changes are in fact saving. The user raised this as they thus believed the changes were not being saved and it was only from testing we realised they were. This is disconcerting for the user.

## Expected Behavior

User should be able to add and remove products and have them save successfully when an order cycle has a schedule attached without an error message being displayed.

## Actual Behaviour

Attempts to save any changes to product lists when a schedule is attached are triggering an error message, making it seem to user that changes are not being saved. This is only happening when a schedule is attached.

## Steps to Reproduce

1. Go to Banc Organic's order cycles
2. Select edit on an open order cycle with a schedule attached
3. Add an in stock product to the incoming and outgoing products list, click save _(N.B. christmas box in screen capture below is not in stock, use 'Test product' under Banc Organics producer to see product displayed on shopfront)_
4. Observe 'failed to update order cycle' error triggered
5. Observe in stock product now displaying on shopfront

## Animated Gif/Screenshot

Kapture 2022-06-10 at 11 32 12

## Workaround

Product list changes are indeed being saved, but are triggering the error. Removing the schedules and clicking save, then making product list edits does not trigger the error, which suggests the issue may be with schedules.

## Severity

bug-s4: it's annoying, but you can use it

Assigning as S4 as the concern is that this could mask other issues, i.e. whilst changes are saving despite the error being shown, should there be a 'real' error the user will not be aware of this. It is also unnerving for the user to not have confirmation that changes to a large product list are being saved.

https://github.com/openfoodfoundation/openfoodnetwork/wiki/Bug-severity
-->

## Your Environment

* Version used: UK prod
* Browser name and version: Chrome & mozilla both showing same bug
* Operating System and version (desktop or mobile): Mac & windows

### PR DESCRIPTION
**Pull request by jibees**
#### What? Why?
This is a tough one, I hope the solution is good. 
- Closes #9292

#### What should we test?

The setup I used to _reproduce_ (I hope...) the issue:
 - Having a closing OC with a schedule
Capture d’écran 2023-03-03 à 15 09 39

 - Adding product to this schedule (here `Garlic 200g`)
Capture d’écran 2023-03-03 à 15 10 11

- Delete this variant
- Try to update OC. _Before_ the PR, you should see an error message ; _After_ the PR, the OC must be edited with a success message.

#### Release notes

Changelog Category: User facing changes | Technical changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Review by mkllnk — CHANGES_REQUESTED** [review]
Great. Spec? :smile:

**Comment by jibees** [comment]
> Great. Spec? 😄

:+1: Done with commit a9502abff6

**Review by mkllnk — APPROVED** [review]

File: app/controllers/admin/order_cycles_controller.rb [hunk 1]
```diff
@@ -100,10 +100,10 @@ def update_nil_subscription_line_items_price_estimate(order_cycle)
       order_cycle.schedules.each do |schedule|
         Subscription.where(schedule_id: schedule.id).each do |subscription|
           shop = Enterprise.managed_by(spree_current_user).find_by(id: subscription.shop_id)
+          fee_calculator = OpenFoodNetwork::EnterpriseFeeCalculator.new(shop, order_cycle)
```
Comment (dacook) [inline app/controllers/admin/order_cycles_controller.rb:103]: 🙌

**Review by dacook — APPROVED** [review]
Great fix.

**Comment by drummer83** [comment]
Hi @jibees,
Thanks for this one!

## Test setup to reproduce the issue
- have a customer with a subscription (kunde2@k... @Konrad Producer Hub (Tax)
- have the order cycle closed in the past
- add a product to the subscription ($NaN is displayed as price) and save
- delete this variant from product list
- try to update the order cycle (change date, name, incoming/outgoing, etc.) 💥 
Error: Order could not be saved, but changes are actually saved.

## After the PR
- same steps as above
No error. Date, name, incoming/outgoing products etc. can be updated and changes are successfully saved. The screen is also showing the green confirmation at the bottom.

image

This is just great! 🤗 
No regression was found! Merging! 🎉 

Thanks again!

**Pull request merged**